enhancements/security/bmc-ca-certificate-support.md

```diff
@@ -0,0 +1,288 @@
+the BMCs using the provided CA certificate before the installation begins.
+
+### Drawbacks
+
+- Introducing CA certificate support increases complexity to the installation
+  and management process.
+
+- Creating or deleting the CA certificate ConfigMap in an existing cluster
+  causes the Ironic container to restart, leading to temporary unavailability.
+
+- CA certificate pre-validation performed in installer is only for cluster
+  installation. There is no automatic pre-validation process for CA certificate
+  updates in an existing cluster.
+
+## Design Details
+
+### Test Plan
+- Unit Test
+  - Add UT for CBO.
+
+- E2E test
+  - Adding e2e test during IPI deployment to verify the following:
+    - The certificate specified by the user is correctly mounted under the
+      fixed path of the ironic container.
+    - The address of the certificate in the ironic container can be passed to
+      ironic for verification.
+
+### Graduation Criteria
+
+This work will be GA immediately since there is no phasing possible or planned.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+N/A
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+This enhancement should not affect the Upgrade/Downgrade Strategy.
+
+### Version Skew Strategy
+
+This enhancement requires the support of both `cluster-baremetal-operator`
+and `ironic-image`. The version of ironic-image is controlled by
+`cluster-baremetal-operator`, thus there will be no version skew.
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+1. Before the IPI installation begins, the `installer` will perform a
+   pre-validation using the provided CA certificate. If the validation fails,
+   an error will be reported immediately, indicating which baremetal BMC failed
+   the validation, and the installation will not proceed.
+
+2. If the pre-validation passes, but a certificate validation error occurs
+   during the installation process (e.g., due to the changes to the baremetal
+   BMC certificate or a man-in-the-middle attack), the user will not be able to
+   see the error message directly until timeout. However, if the user notices
+   that the installation duration is abnormal, they can check the `ironic`
+   container for error messages similar to the following:
+
+   ```log
+   ERROR ironic.conductor.utils requests.exceptions.SSLError: HTTPSConnectionPool(host='xxx.xxx.xxx.xxx', port=443): Max retries exceeded with url: /rest/v1/Oem/eLCM/ProfileManagement/BiosConfig (Caused by SSLError(SSLError(524297, '[X509] PEM lib (_ssl.c:4293)')))
+   ```
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+TBD
+
+## Alternatives
+
+An alternative approach could be to add these CA certificates through the
+`additionalTrustBundle`. However, it does not work for IPI control plane
+installation and it is also insecure to mix the CA certificates used for BMC
```

Comment:
Is it?
If you don't trust one of your CAs not to sign some random certificate, then in what sense can you trust it at all?

Reply:
The description here may be a bit misleading, we don't mean that some CAs are not trusted, but that it's not the most secure way. To be honest, we just refered to the first Open Questions of the enhancement [OpenStack CA Certificate Support](https://github.com/openshift/enhancements/blob/master/enhancements/security/openstack-self-signed-certificates.md#how-should-we-pass-the-certificate-to-the-installer), this topic should have been discussed there, and we just refered to the conclusion.

Reply:
It would be useful to link this info in the enhancement.

Reply:
Updated.